### ISSUE TITLE
Allow multiple NetworkLoadBalancedFargateService's to have the same load balancer 

### ISSUE THREAD

**Issue by likeabbas** [aws/aws-cdk#8040]
### The Problem
Currently, there is an issue where two NetworkLoadBalancedFargateServices cannot have the same NLB because the [NetworkLoadBalancedServiceBase assumes it needs to create a listener even when a listeningPort is not passed in](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-ecs-patterns/lib/base/network-load-balanced-service-base.ts#L288).

This may be due to the fact that, currently, [there is no mechanism to view the NetworkListeners attached to the load balancer](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts#L98).

The reason I want two NetworkLoadBalancedFargateServices to share a listener is because I'm setting up a OneBox environment where the OneBox fargate service will share the same VPC, LoadBalancer, and ECS Cluster with the Prod fargate service. The fargate services need to be identical in every every, except for their auto-scaling parameters. 

### Steps to Reproduce  
    // Setup Stack
    // Inside Stack constructor 
    const nlbfs1 = new ecsPatterns.NetworkLoadBalancedFargateService(this, 'Service1', {...});
    new ecsPatterns.NetworkLoadBalancedFargateService(this, 'Service2', {loadBalancer: nlbfs1.loadBalancer});

### Environment

  - **CDK CLI Version:** 1.39.0
  - **Module Version:** 1.39.0
  - **OS:** OSX Mojave
  - **Language:** TypeScript
### The Solution
I have opened a pull request to extend the Network Load Balanced Base Service class to view the listeners on the NLB, and updated the NetworkLoadBalancedFargateService to check for listeners on the listenerPort before adding a listener 
#8039

**Comment by SoManyHs** [comment]
Hi @likeabbas,

Many apologies about the delayed response. I was curious about your use case here -- it's not possible to have more than one target group on an NLB listener, although that is currently a [bug](https://github.com/aws/aws-cdk/issues/11943) on the ELBv2 construct.

I see in your [pull request](https://github.com/aws/aws-cdk/pull/8039) you appear to be actually adding a new listener per service, rather than having two services share the same actual listener. I'm not clear on whether this is what you actually intended, since in your issue description it sounds like you want all aspects of the service to be the same besides autoscaling which suggests to me you'd want the prod and onebox service to both have the same NLB listener port. If that is the case, adding another listener won't work since you cannot specify more than one listener on an NLB with the same port and protocol.

I'm wondering if instead your use case could be solved by having both your services in the same target group?

**Comment by likeabbas** [comment]
Hi @SoManyHs,

To answer the main portion of your question 

>I'm wondering if instead your use case could be solved by having both your services in the same target group?

Yes. I unblocked myself by using this method, however I removed my use of this construct and simply made two Fargate services attached to the same NLB Target Group. My initial attempt of a PR was trying to fix the construct, but it was in my very novice days of understanding CDK. 

I think the ideal solution would be to allow the user to pass an NLB TargetGroup to the construct, wouldn't you say?
